config/config.go

```diff
@@ -238,6 +245,29 @@ func (c Config) FindComponent(name string) *resource.Config {
 	return nil
 }
 
+// SetToCache sets toCache with a marshalled copy of the config passed in.
+func (c *Config) SetToCache(cfg *Config) error {
```

Comment:
[minor] this method is public to accommodate it's usage in `config/watcher_test.go` right? or are we planning to use it other packages as well? If the former, I suggest we make this method private and define a public version of it in `config/export_test.go`. This seems to be [a common pattern that the golang standard library](https://go.dev/src/net/http/export_test.go) uses to export test-only functions that are used across different packages and `_test` modules. 

(This is a general pattern we can use around our code-base to if we like it)

Reply:
I think being public here is fine -- even if for the sake of testing.

Reply:
will leave as is for now

Reply:
> I think being public here is fine -- even if for the sake of testing. Data tried playing games with keeping things private, but publishing selective things for testing and it became a huge mess.

What made it a mess? Was it unclear what methods were _actually_ public vs public for testing only? If so, I think we can clarify that with some simple sign-posts like requiring any test-only method to take a `testing.TB` interface or something like that. But in any case I don't feel too strongly about this. Happy to leave as is.

Reply:
> What made it a mess? Was it unclear what methods were actually public vs public for testing only?

No, worse. The indirection resulted in a bunch of interfaces. And finding out which method was getting called (in testing and* in production) was impossible without knowing how the magic worked.

If I wanted to pick a single file as the starting point, I'd say it's [this file](https://github.com/viamrobotics/rdk/blob/32e80bb545332ec7f6635b5eed4af93491e1fe0e/services/datamanager/builtin/export_test.go).

What is a DMService? What is a ManagerConstructor? Following the ManagerConstructor definition, what is a [datasync.Manager](https://github.com/viamrobotics/rdk/blob/dfb590cba2819d9ebcd06ad04e73872b250e2dc3/services/datamanager/datasync/sync.go#L41)? Is the `*datamanger.builtin.builtin` type a `datamanager.internal.DMService`? Is it a DMService if testing is not enabled? Does that even matter?